src/EditorFeatures/Core/Implementation/LineSeparators/AdornmentManager.cs

```diff
@@ -273,6 +279,16 @@ private void UpdateInvalidSpans()
                 var tagSpans = _tagAggregator.GetTags(changedSpan);
                 foreach (var tagMappingSpan in tagSpans)
                 {
+                    if (projectionSnapshot != null)
+                    {
+                        var point = tagMappingSpan.Span.Start.GetPoint(changedSpan.Snapshot, PositionAffinity.Predecessor);
```

Comment:
I have no idea, rather than changedSpan.Snapshot, can you just pass projectionSnapshot, and do the additional MapFromSourceSnapshot call for you? I honestly have no idea.